### PR TITLE
Extend Rebuild test support

### PR DESCRIPTION
**Pull request by jaredpar**
Expand rebuild test support to cover more permutation of options
including the different VB runtime combinations

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL test only change

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs [hunk 1]
```diff
@@ -105,21 +110,46 @@ void Method()
...
 
-            PermutateOptimizations("hw.cs /target:exe /debug:embedded", "test.exe", null);
-            PermutateOptimizations("lib1.cs /target:library /debug:embedded", "test.dll", null);
+            Add(Permutate(new CommandInfo("hw.cs /target:exe /debug:embedded", "test.exe", null)));
+            Add(Permutate(new CommandInfo("lib1.cs /target:library /debug:embedded", "test.dll", null)));
 
             return list;
 
-            void PermutateOptimizations(string commandLine, string peFileName, string? pdbFileName)
+            IEnumerable<CommandInfo> Permutate(CommandInfo commandInfo)
```
Comment (cston) [inline src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs:118]: >IEnumerable [](start = 12, length = 11)

`static`

File: src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs [hunk 2]
```diff
@@ -167,25 +197,59 @@ End Module
         {
             var list = new List<object?[]>();
 
-            PermutateOptimizations("hw.vb /target:exe /debug:embedded", "test.exe", null);
+            Add(Permutate(new CommandInfo("hw.vb /target:exe /debug:embedded", "test.exe", null)));
 
             return list;
 
-            void PermutateOptimizations(string commandLine, string peFileName, string? pdbFileName)
+            IEnumerable<CommandInfo> Permutate(CommandInfo commandInfo)
```
Comment (cston) [inline src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs:204]: >IEnumerable [](start = 12, length = 11)

Consider adding `static` for each of these local functions.

**Review by cston — APPROVED** [review]

File: src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs (hunk 2 above)
Comment (jaredpar) [reply src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs:204]: They can't be `static` because they manipulate `list`

**Comment by jaredpar** [comment]
About to queue another PR and will get the `static` added to that. Want to unblock this so Rikki can merge it into his tests.

**Pull request merged**